components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java

```diff
@@ -14789,6 +14792,32 @@ public EndpointRegistryInfo getEndpointRegistryByUUID(String registryId) throws
         return null;
     }
 
+    /**
+     * Deletes an Endpoint Registry
+     *
+     * @param registryUUID Registry Identifier(UUID)
+     * @param registryId Registry Identifier
+     * @throws APIManagementException if failed to delete the Endpoint Registry
+     */
+    public void deleteEndpointRegistry(String registryUUID, int registryId) throws APIManagementException {
+        String deleteRegQuery = SQLConstants.DELETE_ENDPOINT_REGISTRY_SQL;
+        String deleteRegEntryQuery = SQLConstants.DELETE_ENDPOINT_REGISTRY_ENTRY_BY_REGISTRY_ID_SQL;
```

Comment:
We don't need this right? since we have 'delete on cascade' ?

Reply:
Ah yes. I ll update it

Reply:
Fixed in **027f192**